src/tcp_server.rs

```diff
@@ -243,44 +238,111 @@ impl TcpServer {
+                                                    "layer-change".to_string(),
+                                                    "current-layer-name".to_string(),
+                                                    "current-layer-info".to_string(),
+                                                    "fake-key".to_string(),
+                                                    "set-mouse".to_string(),
+                                                ];
+                                                let msg = ServerMessage::HelloOk {
+                                                    version,
+                                                    protocol: 1,
+                                                    capabilities,
+                                                };
+                                                match stream.write_all(&msg.as_bytes()) {
+                                                    Ok(_) => {
+                                                        let _ = stream.flush();
+                                                    }
+                                                    Err(err) => {
+                                                        log::error!(
+                                                            "Error writing HelloOk response: {err}"
+                                                        );
+                                                        connections.lock().remove(&addr);
+                                                        break;
+                                                    }
+                                                }
+                                            }
+                                            // Enhanced reload commands with wait/timeout
+                                            ref reload_cmd @ (ClientMessage::Reload { .. }
+                                            | ClientMessage::ReloadNext {
+                                                ..
+                                            }
+                                            | ClientMessage::ReloadPrev {
+                                                ..
+                                            }
+                                            | ClientMessage::ReloadNum {
+                                                ..
+                                            }
+                                            | ClientMessage::ReloadFile {
+                                                ..
+                                            }) => {
+                                                // Extract wait and timeout from command
+                                                let (wait_flag, timeout) = match reload_cmd {
+                                                    ClientMessage::Reload { wait, timeout_ms } => {
+                                                        (*wait, *timeout_ms)
```

Comment:
These lines extracting `wait` and `timeout_ms` can be moved to the outer match from lines 272-284 to reduce lines of code.